src/main/webapp/components/directives/cache/remotestore/remotestore.js

```diff
@@ -0,0 +1,39 @@
+(function () {
+  'use strict';
+
+  var module = angular.module('ispn.directives.cache.remotestore', ['ispn.services.utils']);
+
+  module.directive('remotestore', ['utils', function (utils) {
+      return {
+        restrict: 'E',
+        scope: {
+          data: '=',
+          metadata: '=',
+          initDefaults: '=',
+          readOnly:'='
+        },
+        replace: true,
+        templateUrl: 'components/directives/cache/remotestore/remotestore.html',
+        link: function (scope, element, attrs) {
+
+          if (scope.initDefaults){
+            scope.data = {};
+            scope.data.cache = scope.metadata.cache.default;
```

Comment:
Yes, but each directive has specific fields that need to be initialized. Service would go through field parameter names and initalize fields? Sounds interesting if it could be done.